mtags/src/main/scala-2/scala/meta/internal/pc/SemanticdbTextDocumentProvider.scala

```diff
@@ -15,6 +18,9 @@ class SemanticdbTextDocumentProvider(val compiler: MetalsGlobal) {
       cursor = None
     )
     typeCheck(unit)
+    import semanticdbOps._
+    // This cache is never update in semanticdb and will contain the old source
+    gSourceFileInputCache.remove(unit.source)
```

Comment:
This seems to be caused by the assumption that the plugin would run only in scalac.